views/dissertation.py

```diff
@@ -104,97 +110,95 @@ def dissertations(request):
 
 @login_required
 @user_passes_test(adviser.is_manager)
+@check_for_dissert(autorized_dissert_promotor_or_manager)
 def manager_dissertations_detail(request, pk):
     dissert = dissertation.find_by_id(pk)
     if dissert is None:
         return redirect('manager_dissertations_list')
     person = mdl.person.find_by_user(request.user)
     adv = adviser.search_by_person(person)
-    if adviser_can_manage(dissert,adv):
-        count_dissertation_role = dissertation_role.count_by_dissertation(dissert)
-        count_proposition_role = proposition_role.count_by_dissertation(dissert)
-        proposition_roles = proposition_role.search_by_dissertation(dissert)
-        offer_prop = offer_proposition.get_by_dissertation(dissert)
-
-        if offer_prop is None:
-            return redirect('manager_dissertations_list')
-
-        files = dissertation_document_file.find_by_dissertation(dissert)
-        filename = ""
-        for file in files:
-            filename = file.document_file.file_name
+    count_dissertation_role = dissertation_role.count_by_dissertation(dissert)
+    count_proposition_role = proposition_role.count_by_dissertation(dissert)
+    proposition_roles = proposition_role.search_by_dissertation(dissert)
+    offer_prop = offer_proposition.get_by_dissertation(dissert)
 
-        if count_proposition_role == 0:
-            if count_dissertation_role == 0:
-                justification = "%s %s %s" % ("auto_add_jury", 'PROMOTEUR', str(dissert.proposition_dissertation.author))
+    if offer_prop is None:
+        return redirect('manager_dissertations_list')
+    files = dissertation_document_file.find_by_dissertation(dissert)
+    filename = ""
+    if files:
+        filename = files[-1].document_file.file_name
+
+    if count_proposition_role == 0:
+        if count_dissertation_role == 0:
```

Comment:
Ces deux if peuvent devenir :` if count_proposition_role == 0 and count_dissertation_role == 0:`